corehq/apps/linked_domain/applications.py

```diff
@@ -57,12 +70,22 @@ def get_upstream_app_ids(downstream_domain):
     return list(_get_downstream_app_id_map(downstream_domain))
 
 
-@quickcache(vary_on=['downstream_domain'], skip_arg=lambda _: settings.UNIT_TESTING, timeout=5 * 60)
-def _get_downstream_app_id_map(downstream_domain):
+@quickcache(
+    vary_on=['downstream_domain', 'use_upstream_app_id'],
+    skip_arg=lambda *args, **kwargs: settings.UNIT_TESTING,
```

Comment:
Is this the best approach to handle additional parameters? It's a bit indiscriminate which isn't great. I suppose the alternative is `lambda domain, use_upstream_app_id: settings.UNIT_TESTING`

Reply:
I don't know, but perhaps another reviewer does.

Reply:
@dannyroberts @millerdev any thoughts here on Graham's question?

Reply:
It's a smell to have test code in production code, so any reference to `settings.UNIT_TESTING` in production code is not ideal. Is there a specific set of tests that you want to disable caching for, or is this a thing that is necessary more widely because many tests happen to call this method as a side effect of what they are doing?

If it's a specific set, I'd try to make a test utility that patches out the caching behavior for just those tests. If it's more broad possibly maybe put the patch in `corehq/tests/noseplugins/patches.py`. We certainly don't want a huge accumulation broad/generic patches like that, so it might be worth thinking about why what you are doing here is special and looking at how other test code handles quickcache.

Reply:
Thanks for the response @millerdev. My question was more specifically about structuring the lambda, as the `settings.UNIT_TESTING` was already being used here. This is also a paradigm we use elsewhere in HQ with quickcache, so I'm tempted to say this falls outside of the scope of this ticket.